woocommerce/woocommerce-ios
# Product Creation AI: New tags/categories

Closes: #10841 

## Description

We have only been asking AI to match existing tags/categories. Starting from this PR
- Ask AI to suggest new tags/categories if existing items don't match.
    - We also load tags/categories before generating a product.
- Save the new AI-suggested tags/categories before saving a product.

Internal discussion about the flow - p1696499410470729/1696499095.943969-slack-C03L1NF1EA3

## Testing instructions
- Create a JN site with the Jetpack AI plugin. (Checkbox available on JN home page to install this plugin)
- Login into the app using the JN site
- Switch to the Products tab and select the "+" button to add a new product. 
- Tap "Create a product with AI" option
- Enter a product name and tap "Continue"
- Enter product features in the next screen and tap "Create Product Details"
- Optionally tap on "Set tone and voice" and select a tone.
- The product details should be displayed on the next screen after the loading animation.
- Scroll down and validate the categories/tags are available.
- Save the product as a draft, and the new categories/tags should be saved along with the Product. 

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/524475/c74c72da-d4cc-4005-9ff8-68e806ea3551



---
